runners/samza/build.gradle

```diff
@@ -53,21 +54,28 @@ dependencies {
   compile library.java.slf4j_api
   compile library.java.joda_time
   compile library.java.commons_io
-  compile library.java.commons_lang3
-  compile library.java.args4j
-  compile "javax.servlet:javax.servlet-api:3.1.0"
-  compile "io.dropwizard.metrics:metrics-core:3.1.2"
   compile "org.rocksdb:rocksdbjni:5.7.3"
-  compile "org.apache.commons:commons-collections4:4.0"
   compile "org.scala-lang:scala-library:2.11.8"
   compile "org.apache.samza:samza-api:$samza_version"
   compile "org.apache.samza:samza-core_2.11:$samza_version"
   compile "org.apache.samza:samza-kafka_2.11:$samza_version"
   compile "org.apache.samza:samza-kv_2.11:$samza_version"
+  compile project(":sdks:java:expansion-service")
+  permitUnusedDeclared "org.apache.samza:samza-kafka_2.11:1.5.0"
```

Comment:
I'm guessing many/all of these `permitUnusedDeclared` dependencies should be `runtimeOnly` instead of `compile`+`permitUnusedDeclared`. But I will let someone with more knowledge of the Samza runner say for sure.

Reply:
Makes sense. Most of them are used in runtime. I am not sure why the samza-kv_2.11 module needs to be permitUnusedDeclared, as it contains Samza's state backend api and we use it in the runner's state internals impl.